### DISCUSSION TITLE
Towards passwordless unlocking

### DISCUSSION THREAD

**Discussion by miquecg** [keepassxreboot/keepassxc#8085] [Q&A] [answered]
Excuse my ignorance if I'm asking something that doesn't make sense.

Do you think it would be possible for KeePassXC to allow database unlock with either a hardware device or a password?

I'm thinking of a setup where you can unlock a database by simply having a [SmartCard](https://github.com/keepassxreboot/keepassxc/issues/255) plugged in, while also keeping a long password securely stored mostly as a fallback. As I understand it, right now hardware keys are an additional layer of security on top of a master password.

I'd like to achieve something like that, having convenience without completely defeating the purpose of using a password manager.

**Comment by droidmonkey** [comment, marked as answer]
The keepass standard does not support having multiple credential sets per database. However, you can achieve this same behavior by using Windows Hello and a Yubikey. You can also just use a Yubikey without any password, or a very simple password.

**Reply by miquecg** [reply]
Ok, so when #255 is implemented one could use a SmartCard without any password?

**Reply by miquecg** [reply]
> The keepass standard does not support having multiple credential sets per database.

I was thinking about one possibility to resemble that of multiple credentials. Use a hardware device to unlock a database which in turn then auto opens another one with a very long password. So essentially I'm protecting the master password behind a hardware key, also very convenient to use, but having the chance to always resort to the password, for instance when recovering database from a backup on another device.

About my previous question, do you think PKCS#11 support will land shortly?

**Reply by droidmonkey** [reply]
Yes I still want to implement pkcs#11 support. The auto open idea is really good and pretty much solves the problem.
